internal/server/repository_worker.go

```diff
@@ -197,6 +197,11 @@ func (r *Repository) ListWorkers(ctx context.Context, scopeIds []string, opt ...
 		return nil, errors.New(ctx, errors.InvalidParameter, op, fmt.Sprintf("unknown worker type %v", opts.withWorkerType))
 	}
 
+	if opts.withExcludeShutdownWorkers {
```

Comment:
Consider flipping this to something like `withActiveWorkers`. I believe the only place using this option is the target service, but for that you likely want to only pull in active workers and not also pull in unknown workers.

There is of course a backwards compat issue there...I'm not sure if we want to make a clean "GA requires GA" break or not. So maybe something that should be discussed more broadly. Alternately, the query could get a bit more complex: active could indicate workers _with_ a release version but in active state (not unknown or shutdown) or any workers _without_ a release version.

Either way I do think excluding any unknown state workers if possible is probably for the best.

Reply:
Oh, or maybe remove the WithExcludeShutdownWorkers and just use the WithOperationalState option instead, and then put in the logic that uses release version as a switch for backwards compat as you do elsewhere?